tensorboard/plugins/interactive_inference/tf_interactive_inference_dashboard/tf-interactive-inference-dashboard.html

```diff
@@ -6227,20 +6296,160 @@ <h2>Show similarity to selected datapoint</h2>
                 const attribSparseKeyKey =
                   attributionPrefix + keys[j] + attributionSparseKeySuffix;
                 datapoint[attribSparseKeyKey] = attribs[keys[j]][0];
-                const attribSparseValueKey =
+                attribKey =
                   attributionPrefix + keys[j] + attributionSparseValueSuffix;
-                datapoint[attribSparseValueKey] = attribs[keys[j]][1];
+                datapoint[attribKey] = attribs[keys[j]][1];
               } else {
-                const attribKey = attributionPrefix + keys[j];
                 datapoint[attribKey] = attribs[keys[j]];
               }
+              let newAttributions = Array.isArray(datapoint[attribKey])
+                ? datapoint[attribKey]
+                : [datapoint[attribKey]];
+              for (
+                let attribIdx = 0;
+                attribIdx < newAttributions.length;
+                attribIdx++
+              ) {
+                if (newAttributions[attribIdx] > this.maxAttribution) {
+                  this.maxAttribution = newAttributions[attribIdx];
+                }
+                if (newAttributions[attribIdx] < this.minAttribution) {
+                  this.minAttribution = newAttributions[attribIdx];
+                }
+              }
             }
             this.set(`visdata.${idx}`, datapoint);
           }
+          // Make min/max symmetric around 0 so that attribution visualization scales
+          // for negative and positive attributions are the same, for visual
+          // consistency.
+          if (
+            this.minAttribution < 0 &&
+            this.maxAttribution > Math.abs(this.minAttribution)
+          ) {
+            this.minAttribution = -1 * this.maxAttribution;
+          } else if (
+            this.maxAttribution > 0 &&
+            Math.abs(this.minAttribution) > this.maxAttribution
+          ) {
+            this.maxAttribution = -1 * this.minAttribution;
+          }
           this.setAttributions();
+          this.createAttributionLegend();
           this.refreshDive_();
         },
 
+        // Create and display the attribution legend.
+        createAttributionLegend: function() {
+          this.attributionColorScale = d3
+            .scaleLinear()
+            .domain([this.minAttribution, 0, this.maxAttribution])
```

Comment:
does this create an issue when minAttribution is positive or maxAttribution is positive?

Reply:
good point. fixed